honeycomb/honeycomb.go

```diff
@@ -51,6 +51,8 @@ type Config struct {
 	// APIHost is the hostname for the Honeycomb API server to which to send
 	// these events. default: https://api.honeycomb.io/
 	APIHost string
+	// UserAgent will set the user agent used by the exporter
+	UserAgent string
```

Comment:
If we adopt the proposed style from #54, we could replace this with a `WithUserAgent` function.

Reply:
Yup! I'll be looking at that closely this week btw, looks like a good improvement.